specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "./azure_chat_extensions.tsp";
+import "./common.tsp";
+import "./functions.tsp";
+import "./tools.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+#suppress "@azure-tools/typespec-autorest/union-unsupported" "External API shape is defined in OpenAPI 3.0 as oneOf."
+alias ChatMessageContent = string | ChatMessageContentItem[];
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("An abstract representation of a structured content item within a chat message.")
+@discriminator("type")
+model ChatMessageContentItem {
+  @doc("The discriminated object type.")
+  type: string;
```

Comment:
The type of of `type` (gosh) seems to be refered as `chatCompletionRequestMessageContentPart` in the yaml. Maybe we should include the enum as well, to have the representation match more strictly and not leaving it so open eneded? Not entirely sure what are the ramifications of using `@discriminator` with a `string` instead of an `enum`

Reply:
Per the comment a bit later, I'd like to bring up the holistic topic of whether we should strongly type all discriminators as named enums. Then we can switch everything all at once.